### PR TITLE
Create basic authentication functions in Server using bcrypt as hashi…

### LINKED ISSUE #10: Create basic authentication functions in Server using bcrypt as hashing library
## Context 

Our app will have authentication features that will interact with our Server.

## Problem 

The backend system running in the server needs functions and APIs to communicate with the database to perform authentication.

## Solution: 

- [x] Add bcrypt dependency
- [x] Create API for registration
- [x] Create API for login

### PR DESCRIPTION
**Pull request by jyaneshseeam**
# Pull Request Name
Implement login and signup/registration functions

## Source 
Closes #10

## Description 
Adds bycrypt as dependency and functions for login and user sign up now available

### Summary of PR  

Updates package.json to add bycrypt dependency and updates to server.js to add login and signup functionality and to accept json res and req + request.rest file created to use REST CLIENT extension for API testing

### Purpose 
To perform backend authentication 

## Review and Testing Steps
[ ] Pull the latest branch
[ ] Run npm install
[ ] Add Rest client extension to VS code
[ ] Run npm run devStart
[ ] open request.rest file
[ ] Play around with some requests

## Additional Information (If applicable)
N/A

### Notes
Make sure you have Rest.client extension added to vs code, else you can use a tool like POSTMAN
Routes are 
localhost:3000/users/register (to register new user)
localhost:3000/users/login ( to login)
localhost:3000/users returns list of users registered (for admin purposes, will be later removed)
### Resources 
Watched this [tutorial on how to use bcrypt and implement basic login functions](https://www.youtube.com/watch?v=Ud5xKCYQTjM)

### CODE REVIEW

**Comment by DemiAdisa** [comment]
Where is the request.rest file located?

**Comment by jyaneshseeam** [comment]
> Where is the request.rest file located?

@DemiAdisa 
It is found at : 
app/server/request.rest

**Comment by DemiAdisa** [comment]
> > Where is the request.rest file located?
> 
> 
> 
> @DemiAdisa 
> 
> It is found at : 
> 
> app/server/request.rest

I don't see it anywhere, but I may not have ran git pull properly image

**Review by DemiAdisa — APPROVED** [review]
Runs perfectly fine, good job!

File: app/server/request.http [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+GET http://localhost:3000/users
+
+###
+
+//post request to register new user with username and password in json
+POST http://localhost:3000/users/register
+Content-Type: application/json
+
+{
+"userName": "Roshan",
```
Comment (sontushkpaul) [inline app/server/request.http:11]: Do not add any credentials in github. Share it in Google docs or discord

File: app/server/server.js [hunk 2]
```diff
@@ -1,19 +1,77 @@
...
+  }
+})
+
+
+
+
+
 //Attempt to connect to MongoDB Atlas
 mongoose.connect(
   "mongodb+srv://mindjogg_the_app:Mindj0Gg123@mindjoggcluster.e2dzp.mongodb.net/myFirstDatabase?retryWrites=true&w=majority",
```
Comment (sontushkpaul) [inline app/server/server.js:78]: put it in .env file. It is a security breach. Good for now change it later.

**Review by sontushkpaul — APPROVED** [review]

**Review by ja-chisholm — APPROVED** [review]
Aside from what Sontush pointed out, all looks good

**Pull request merged**